collections/binary_search_tree.ts

```diff
@@ -236,14 +236,11 @@ export class BinarySearchTree<T> implements Iterable<T> {
   }
 
   protected removeNode(
```

Comment:
It looks like currently this function previously either returned the removeNode or the successorNode if it found the node. The reason was so that the caller would know whether or not a node was found/removed. Now that we always pass in a node that was found, I think this function can be changed to instead only return the successorNode.

The remove function on the RedBlackTree doesn't need to get the removedNode, it only needs the successorNode so that it can rebalance the tree.

Reply:
Added a comment to reflect my findings. Returned node should be the node physically removed from the tree, and not the successor node. It seems like `RedBlackTree` does need the removed node.